File: 0242-valid-anagram/0242-valid-anagram.cpp
class Solution {
public:
    bool isAnagram(string s, string t) {
        int n = (int)s.size(), m = (int)t.size();
        if (n != m)
            return false;
        int seen[26];
        for (char ch : s)
            seen[ch-'a']++;
        for (char ch : t){
            if (--seen[ch-'a'] < 0)
                return false;
        }
        return true;
    }
};

File: 0242-valid-anagram/0242-valid-anagram.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        m,n = len(s),len(t)
        if m != n: return False
        count = [0] * 26
        for ch in s:
            count[ord(ch)-ord('a')] += 1
        for ch in t:
            count[ord(ch)-ord('a')] -= 1
            if count[ord(ch)-ord('a')] < 0:
                return False
        return True